frontend/appflowy_flutter/lib/ai/service/ai_model_state_notifier.dart

```diff
@@ -0,0 +1,174 @@
+import 'package:appflowy/ai/ai.dart';
+import 'package:appflowy/generated/locale_keys.g.dart';
+import 'package:appflowy/plugins/ai_chat/application/ai_model_switch_listener.dart';
+import 'package:appflowy/workspace/application/settings/ai/local_llm_listener.dart';
+import 'package:appflowy_backend/dispatch/dispatch.dart';
+import 'package:appflowy_backend/log.dart';
+import 'package:appflowy_backend/protobuf/flowy-ai/entities.pb.dart';
+import 'package:appflowy_result/appflowy_result.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:protobuf/protobuf.dart';
+import 'package:universal_platform/universal_platform.dart';
+
+typedef OnModelStateChangedCallback = void Function(AiType, bool, String);
+typedef OnAvailableModelsChangedCallback = void Function(
+  List<AiModel>,
+  AiModel?,
+);
+
+class AIModelStateNotifier {
```

Comment:
I don't know why we have lib/ai directly. why not just put in the chat directory. we can move service to application folder under chat.

Reply:
I put it like this because the AI writer is an editor plugin in the document plugin, while AI chat is it's own chat plugin. And then, they use shared components in lib/ai. Should I change this?

Reply:
I can't answer you how to do it now. I will take a look later.